src/ui/ui_i18n/README.md

```diff
@@ -0,0 +1,169 @@
+
+The translation `service` will provide the following methods (names can be changed):
+- `addMessages(messages: object, [locale: string])` - provides a way to register
+translations with the library
+- `getMessages()` - returns messages for the current language
+- `locale(locale: string)` - tells the library which language to use by given
+language key
+- `locale()` - returns the current locale 
+- `defaultLocale(locale: string)` - tells the library which language to fallback
+when missing translations
+- `defaultLocale()` - returns the default locale
+- `defineFormats(formats: object)` - supply a set of options to the underlying formatter
+- `translate({id: string, values: object, defaultMessage: string})` – translate message by id
+
+The translation `filter` will look like this:
+```
+{{'translationId' | i18n[:{ values: object, defaultMessage: string }]}}
+```
+
+Where:
+- `translationId` - translation id to be translated
+- `values` - values to pass into translation
+- `defaultMessage` - will be used unless translation was successful
+
+The translation `directive` will look like this:
+```
+<ANY
+  i18n="{string}"
+  [i18n-values="{object}"]
+  [i18n-default-message="{string}"]
+></ANY>
+```
+
+Where:
+- `i18n` - translation id to be translated
+- `i18n-values` - values to pass into translation
+- `i18n-default-message` - will be used unless translation was successful
+
+## Localization files
+
+Localization files will have [JSON5](https://github.com/json5/json5) format,
+so single and multi-line comments are allowed. It can help to understand
+which section of application localization key is used for. Also `namespaces`
+will be used in order to simplify message location search. For example, if
+we are going to translate the title of `/management/sections/objects/_objects.html`
+file, we should use message path like this: `'MANAGEMENT.OBJECTS.TITLE'`.
+
+Each Kibana plugin will have a separate folder with translation files located at
+```
+src/core_plugins/{plugin_name}/translations/{locale}.json
+```
+
+where `locale` is [ISO 639 language code](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes).
+
+For example:
+```
+src/core_plugins/kibana/translations/en.json
+```
+
+When a new translation file is added, you have to register this file into
+`uiExports.translations` array of plugin constructor parameters. For example:
+```
+export default function (kibana) {
+  return new kibana.Plugin({
+    uiExports: {
+      translations: [
+        resolve(__dirname, './translations/en.json'),
+      ],
+      ...
+    },
+    ...
+  });
+}
+```
+
+Localization files are loaded in `src/ui/public/chrome/api/angular.js` file and
+provided to angular using `i18nProvider`:
+```
+uiModules.get('kibana').config(function (i18nProvider) {
+  i18nProvider.addMessages(chrome.getTranslations());
+});
+```
+
+After that we are able to provide translation messages to react `IntlProvider`:
```

Comment:
I'm a bit confused here too. We jumped from Angular to React suddenly. Should this actually be a separate section? Or are they connected and I just don't have context?